### PR TITLE
CPU usage by category data

### PR DESCRIPTION
**Pull request by XrosHeart05**
CPUs usage by category data in percentage values

### CODE REVIEW

File: cpu/cpu.sh [hunk 1]
```diff
@@ -1,14 +1,166 @@
...
+# echo "CPU Usage: $cpu"
+# echo "CPUs: $cpus"
+# echo "Sum: $sum"
+# echo "Percentage: $percentage"
+# echo "One top register: $just_one_register_top_with_active_process"
+# echo "//////////////////////////////"
+# echo "//////////////////////////////"
+# echo "Test 1: $only_top"
+# echo "CPUs 2: $cpus2"
+# echo "Test 2: $test2"
```
Comment (tacho090) [inline cpu/cpu.sh]: delete commented code

File: cpu/cpu_by_core.sh [hunk 2]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+
+usa=$(grep -w 'cpu' /proc/stat | awk '{usage=($2+$3+$4)*100/($2+$3+$4+$5+$6+$7+$8)} END {print usage"%"}')
+
+echo "$usa"
```
Comment (tacho090) [inline cpu/cpu_by_core.sh]: is this file required?

File: cpu/cpu.sh [hunk 3]
```diff
@@ -1,14 +1,166 @@
...
+  print_graph_bar "Niced processes (low priority) usage:  " $usage_cpu_low_priority $unusage_cpu_low_priority
+  print_graph_bar "Inactive CPU usage:                    " $usage_cpu_idle $unusage_cpu_idle
+  print_graph_bar "I/O waiting CPU usage:                 " $usage_cpu_iowait $unusage_cpu_iowait
+  print_graph_bar "Hardware interruptions CPU usage:      " $usage_cpu_hardware_i $unusage_cpu_hardware_i
+  print_graph_bar "Software interruptions CPU usage:      " $usage_cpu_software_i $unusage_cpu_software_i
+  print_graph_bar "Stolen tasks CPU usage:                " $usage_cpu_steal $unusage_cpu_steal
+  echo -e "=================\n"
+
+  sleep 1
+done
```
Comment (tacho090) [inline cpu/cpu.sh:141]: JSON parse function missing

**Review by tacho090 — CHANGES_REQUESTED** [review]

File: cpu/cpu.sh (hunk 1 above)
Comment (XrosHeart05) [reply cpu/cpu.sh]: Commented code deleted in this file

File: cpu/cpu_by_core.sh (hunk 2 above)
Comment (XrosHeart05) [reply cpu/cpu_by_core.sh]: The functionality of this file has been updated and even renamed to

**Comment by XrosHeart05** [comment]
New features added; data input to DynamoDB consuming stream data

File: cpu/cpu.sh (hunk 3 above)
Comment (tacho090) [reply cpu/cpu.sh:141]: solved

**Review by tacho090 — APPROVED** [review]
